amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -116,25 +117,37 @@ public TaskResult call() {
-                if (reason == ShutdownReason.SHARD_END) {
+                final Lease currentShardLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
+                if (localReason == ShutdownReason.SHARD_END) {
                     // Create new lease for the child shards if they don't exist.
                     // We have one valid scenario that shutdown task got created with SHARD_END reason and an empty list of childShards.
                     // This would happen when KinesisDataFetcher(for polling mode) or FanOutRecordsPublisher(for StoS mode) catches ResourceNotFound exception.
                     // In this case, KinesisDataFetcher and FanOutRecordsPublisher will send out SHARD_END signal to trigger a shutdown task with empty list of childShards.
                     // This scenario could happen when customer deletes the stream while leaving the KCL application running.
-                    final Lease currentShardLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
                     Validate.validState(currentShardLease != null,
-                            "%s : Lease not owned by the current worker. Leaving ShardEnd handling to new owner.",
-                            leaseKeyProvider.apply(shardInfo));
-                    final LeasePendingDeletion leasePendingDeletion = new LeasePendingDeletion(streamIdentifier,
-                            currentShardLease, shardInfo);
-
-                    if (!CollectionUtils.isNullOrEmpty(childShards)) {
-                        createLeasesForChildShardsIfNotExist();
-                        updateLeaseWithChildShards(currentShardLease);
+                                        "%s : Lease not owned by the current worker. Leaving ShardEnd handling to new owner.",
+                                        leaseKeyProvider.apply(shardInfo));
+                    try {
+                        if (!CollectionUtils.isNullOrEmpty(childShards)) {
+                            createLeasesForChildShardsIfNotExist();
+                            updateLeaseWithChildShards(currentShardLease);
+                        }
+                    } catch (InvalidStateException e) {
+                        // If invalidStateException happens, it indicates we are missing childShard related information.
+                        // In this scenario, we should shutdown the shardConsumer with LEASE_LOST reason to allow other worker to take the lease and retry getting
+                        // childShard information in the processTask.
+                        localReason = ShutdownReason.LEASE_LOST;
+                        dropLease();
+                        log.warn("Shard " + shardInfo.shardId() + ": Exception happened while shutting down shardConsumer with LEASE_LOST reason. " +
+                                         "Dropping the lease and shutting down shardConsumer using ZOMBIE reason. Exception: ", e);
```

Comment:
1. We need leaseKey and not ShardId, as we can be in multistreaming mode
2. Exception happened while shutting down shardConsumer with ~LEASE_LOST~ SHARD_END reason
3. Dropping the lease and shutting down shardConsumer using ~ZOMBIE~ LEASE_LOST reason
4. Use placeholder while logging instead of directly using + operastor

Reply:
Addressed